transports/tls/tls.go

```diff
@@ -2,27 +2,120 @@ package tls
 
 import (
 	"crypto/tls"
+	"crypto/x509"
+	"io/ioutil"
+	"log"
 	"net"
+	"os"
+	"path/filepath"
 
 	"github.com/gliderlabs/logspout/adapters/raw"
 	"github.com/gliderlabs/logspout/router"
 )
 
+// default CA and certificate path
+const (
+	CAPath   = "/mnt/ca/"
+	CertPath = "/mnt/cert/"
+)
+
 func init() {
 	router.AdapterTransports.Register(new(tlsTransport), "tls")
 	// convenience adapters around raw adapter
 	router.AdapterFactories.Register(rawTLSAdapter, "tls")
 }
 
+func getopt(name, dfault string) string {
+	value := os.Getenv(name)
+	if value == "" {
+		value = dfault
+	}
+	return value
+}
+
 func rawTLSAdapter(route *router.Route) (router.LogAdapter, error) {
 	route.Adapter = "raw+tls"
 	return raw.NewRawAdapter(route)
 }
 
+func getCAs(path string) *x509.CertPool {
+	cafiles, err := ioutil.ReadDir(path)
+	if err != nil {
+		return nil
+	}
+	capool := x509.NewCertPool()
+	// TODO: don't khow how to check pool is empty
+	empty := true
+	for _, f := range cafiles {
+		if !f.IsDir() {
+			cacert, err := ioutil.ReadFile(path + f.Name())
+			if err != nil {
+				log.Printf("Can't read CA certificate %v: %v", path+f.Name(), err)
+				continue
+			}
+			ok := capool.AppendCertsFromPEM(cacert)
+			if !ok {
+				log.Printf("Bad CA certificate %v", path+f.Name())
+			}
+			empty = !ok && empty
+		}
+	}
+
+	if empty {
+		capool = nil
+	}
+
+	return capool
+}
+
+func getCertificates(path string) []tls.Certificate {
+	var certs []tls.Certificate
+	certfiles, err := ioutil.ReadDir(path)
+	if err != nil {
+		return certs
+	}
+	for _, f := range certfiles {
```

Comment:
I think this reads a little better (with names changed to match tls.LoadX509KeyPair docs because why not):
```go
for _, f := range certfiles {
	ext := filepath.Ext(f.Name())
	if ext != ".crt" && ext != ".cert" {
		continue
	}

	certFile := filepath.Join(path, f.Name())
	keyFile := certFile[:len(certFile)-len(ext)] + ".key"

	cert, err := tls.LoadX509KeyPair(certFile, keyFile)
	if err != nil {
		continue
	}

	certs = append(certs, cert)
}
```

I'm not sure the os.Stat of keyFile is necessary considering there's no sanitation of the `f`'s type (could be another dir with .crt extension, even if that would be very weird) and LoadX509KeyPair will return an error if its not a file.